helix-term/src/keymap.rs

```diff
@@ -419,7 +305,8 @@ impl Keymaps {
     /// sticky node is in use, it will be cleared.
     pub fn get(&mut self, mode: Mode, key: KeyEvent) -> KeymapResult {
         // TODO: remove the sticky part and look up manually
-        let keymap = &self.map[&mode];
+        let keymaps = self.map().clone();
```

Comment:
Why the need to clone here? I imagine this is cloning the entire keymap which might be very heavy.

Reply:
Good call, I thought this was an arc clone but it was not. I can change this to `&*self.map()`